iotdb-core/datanode/src/main/java/org/apache/iotdb/db/utils/ModificationUtils.java

```diff
@@ -404,16 +403,18 @@ public static List<ModEntry> sortAndMerge(List<ModEntry> modifications) {
   }
 
   public static boolean isDeviceDeletedByMods(
-      Collection<ModEntry> currentModifications, ITimeIndex currentTimeIndex, IDeviceID device)
+      final Collection<ModEntry> currentModifications,
+      final ITimeIndex currentTimeIndex,
+      final IDeviceID device)
       throws IllegalPathException {
-    if (currentTimeIndex == null) {
-      return false;
-    }
-    Optional<Long> startTime = currentTimeIndex.getStartTime(device);
-    Optional<Long> endTime = currentTimeIndex.getEndTime(device);
-    if (startTime.isPresent() && endTime.isPresent()) {
-      return isAllDeletedByMods(currentModifications, device, startTime.get(), endTime.get());
-    }
-    return false;
+    return isAllDeletedByMods(
```

Comment:
Will the logic be different from the former version when either startTime or endTime is not present?